modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/states/WaitForInit.kt

```diff
@@ -68,7 +68,11 @@ data object WaitForInit : ChannelState() {
                     is ChannelStateWithCommitments -> cmd.state.commitments.active.mapNotNull { commitment ->
                         when (val fundingStatus = commitment.localFundingStatus) {
                             is LocalFundingStatus.UnconfirmedFundingTx -> fundingStatus.signedTx
-                            is LocalFundingStatus.ConfirmedFundingTx -> null
+                            is LocalFundingStatus.ConfirmedFundingTx -> when (fundingStatus.shortChannelId) {
+                                // If the short_channel_id isn't correctly set, we fetch the funding transaction to update it.
+                                ShortChannelId(0) -> fundingStatus.signedTx
+                                else -> null
+                            }
```

Comment:
I see the intention, but I don't get how it works. AFAICT we only call `ChannelAction.Blockchain.PublishTx` on those txs. Shouldn't we put a `WatchConfirm` or something?